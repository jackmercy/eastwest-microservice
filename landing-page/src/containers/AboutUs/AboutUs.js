import { memo, useCallback, useMemo, useState } from 'react';
import Link from 'next/link';
import { useIntl } from 'react-intl';
import CountUp from 'react-countup';
import VisibilitySensor from 'react-visibility-sensor';

import { scrollToElement } from '../../utils';
import { Layout, Header, Footer, Content } from '../components';
import { DEFAULT_SEO } from '../../../configs';
import { Button, SEO } from '../../components';
import MenuScrollableLayout from '../Careers/components/MenuScrollableLayout';

import useStyles from './aboutUs.styles';

const AboutUs = () => {
  const intl = useIntl();
  const classes = useStyles();
  const ids = useMemo(() => ['about-company', 'history', 'our-team'], []);
  const menuConfigs = useMemo(
    () => [
      {
        id: ids[0],
        label: intl.formatMessage({ id: 'ABOUT_US.DETAILS.MENU.01' }),
      },
      {
        id: ids[1],
        label: intl.formatMessage({ id: 'ABOUT_US.DETAILS.MENU.02' }),
      },
      {
        id: ids[2],
        label: intl.formatMessage({ id: 'ABOUT_US.DETAILS.MENU.03' }),
      },
    ],
    [intl],
  );
  const companyInfo = useMemo(
    () => [
      {
        key: 'founded',
        value: 2018,
        countUpConfigs: { prefix: '', suffix: '' },
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.FOUNDED',
        }),
      },
      {
        key: 'talents',
        value: 800,
        countUpConfigs: { prefix: '', suffix: '+' },
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.TALENTS',
        }),
      },
      {
        key: 'projects',
        value: 140,
        countUpConfigs: { prefix: '', suffix: '+' },
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.PROJECTS',
        }),
      },
      {
        key: 'raised',
        value: 1.5,
        countUpConfigs: {
          prefix: '$',
          suffix: 'B+',
          decimals: 1,
          decimal: '.',
        },
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.RAISED',
        }),
      },
    ],
    [intl],
  );
  const historyInfo = useMemo(
    () => [
      {
        year: '2018',
        descriptions: [
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2018.TEXT.1' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2018.TEXT.2' }),
        ],
      },
      {
        year: '2019',
        descriptions: [
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2019.TEXT.1' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2019.TEXT.2' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2019.TEXT.3' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2019.TEXT.4' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2019.TEXT.5' }),
        ],
      },
      {
        year: '2020',
        descriptions: [
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.1' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.2' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.3' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.4' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.5' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.6' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.7' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.8' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2020.TEXT.9' }),
        ],
      },
      {
        year: '2021',
        descriptions: [
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2021.TEXT.1' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2021.TEXT.2' }),
          intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.2021.TEXT.3' }),
        ],
      },
    ],
    [intl],
  );
  const members = useMemo(
    () => [
      {
        name: 'Rabih Souk',
        role: 'Chief Executive officer',
        avatar: '/static/images/about-us/member-1.png',
      },
      {
        name: 'Duyen Tran',
        role: 'Director of Operations',
        avatar: '/static/images/about-us/member-2.png',
      },
      {
        name: 'Tuan Nguyen',
        role: 'Product Design Manager',
        avatar: '/static/images/about-us/member-5.png',
      },
      {
        name: 'Hai Bui',
        role: 'Software engineer',
        avatar: '/static/images/about-us/member-7.png',
      },
      {
        name: 'Hung Le',
        role: 'Software engineer',
        avatar: '/static/images/about-us/member-8.png',
      },
      {
        name: 'Hanh Van',
        role: 'Junior talent Manager',
        avatar: '/static/images/about-us/member-9.png',
      },
    ],
    [],
  );
  const [isVisible, setVisibility] = useState(false);

  const handleScrollToId = useCallback((e, id) => {
    e.preventDefault();
    window.history.pushState({}, '', `/company#${id}`);
    scrollToElement(id);
  }, []);

  const handleVisibilityChange = (status) => {
    if (status) {
      setVisibility(status);
    }
  };

  return (
    <Layout>
      <SEO {...DEFAULT_SEO.PAGES.ABOUT_US} />
      <Header />
      <Content className={classes.teaser}>
        <h1 className={classes.teaserTitle} data-test="about-us-teaser-title">
          {intl.formatMessage(
            { id: 'ABOUT_US.TEASER.TITLE' },
            {
              span: (...chunks) => (
                <span className={classes.primary}>{chunks.join('')}</span>
              ),
            },
          )}
        </h1>
        <p
          className={classes.teaserDescription}
          data-test="about-us-teaser-description"
        >
          {intl.formatMessage({ id: 'ABOUT_US.TEASER.DESCRIPTION' })}
        </p>
      </Content>
      <div className={classes.coverImage} data-test="about-us-teaser-image" />
      <Content className={classes.menuScrollableLayout}>
        <MenuScrollableLayout menuConfigs={menuConfigs}>
          <Link href={`/company#${ids[0]}`}>
            <h1
              id={ids[0]}
              role="presentation"
              className={classes.introduceTitle}
              onClick={(e) => handleScrollToId(e, ids[0])}
            >
              {intl.formatMessage(
                { id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.TITLE' },
                {
                  span: (...chunks) => (
                    <span className={classes.primary}>{chunks.join('')}</span>
                  ),
                },
              )}
            </h1>
          </Link>
          <p className={classes.aboutText}>
            {intl.formatMessage({
              id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.TEXT.01',
            })}
          </p>
          <div
            className={classes.aboutCompany}
            data-test="about-us-details-about-company"
          >
            {companyInfo.map((item) => (
              <div key={item.key} className={classes.aboutCompanySummary}>
                <h2 className={classes.aboutCompanyValue}>
                  <CountUp
                    end={item.value}
                    start={isVisible ? 0 : null}
                    {...item.countUpConfigs}
                  >
                    {({ countUpRef }) => (
                      <VisibilitySensor
                        onChange={handleVisibilityChange}
                        delayedCall
                      >
                        <span ref={countUpRef} />
                      </VisibilitySensor>
                    )}
                  </CountUp>
                </h2>
                <p className={classes.aboutCompanyLabel}>{item.label}</p>
              </div>
            ))}
          </div>
          <p className={classes.aboutText}>
            {intl.formatMessage({
              id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.TEXT.02',
            })}
          </p>
          <p className={classes.aboutText}>
            {intl.formatMessage({
              id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.TEXT.03',
            })}
          </p>
          <Link href={`/company#${ids[1]}`}>
            <h2
              id={ids[1]}
              role="presentation"
              className={classes.introduceTitle}
              onClick={(e) => handleScrollToId(e, ids[1])}
            >
              {intl.formatMessage({ id: 'ABOUT_US.DETAILS.HISTORY.TITLE' })}
            </h2>
          </Link>
          <div
            className={classes.historyCompany}
            data-test="about-us-details-history-company"
          >
            {historyInfo.map((item) => (
              <div
                key={`history-${item.year}`}
                className={classes.historySummary}
              >
                <div className={classes.historyYear}>
                  <span className={classes.historyMark} />
                  {item.year}
                </div>
                <ul className={classes.historyDescriptions}>
                  {item.descriptions.map((description, index) => (
                    <li
                      // eslint-disable-next-line react/no-array-index-key
                      key={`history-${item.year}-${index}`}
                      className={classes.historyItem}
                    >
                      <span className={classes.historyMark} />
                      {description}
                    </li>
                  ))}
                </ul>
              </div>
            ))}
          </div>
          <Link href={`/company#${ids[2]}`}>
            <h2
              id={ids[2]}
              role="presentation"
              className={classes.introduceTitle}
              onClick={(e) => handleScrollToId(e, ids[2])}
            >
              {intl.formatMessage(
                { id: 'ABOUT_US.DETAILS.OUR_TEAM.TITLE' },
                {
                  span: (...chunks) => (
                    <span className={classes.primary}>{chunks.join('')}</span>
                  ),
                },
              )}
            </h2>
          </Link>
          <div
            className={classes.members}
            data-test="about-us-details-company-members"
          >
            {members.map((member) => (
              <div key={member.name} className={classes.member}>
                <div
                  className={classes.memberAvatar}
                  style={{ backgroundImage: `url(${member.avatar})` }}
                />
                <p className={classes.memberName}>{member.name}</p>
                <p className={classes.memberRole}>{member.role}</p>
              </div>
            ))}
          </div>
        </MenuScrollableLayout>
      </Content>
      <div className={classes.starting}>
        <h2
          className={classes.startingTitle}
          data-test="about-us-starting-title"
        >
          {intl.formatMessage({ id: 'ABOUT_US.START_WITH_COMPANY.TITLE' })}
        </h2>
        <Link href="/#create-project-form" passHref>
          <Button
            tag="a"
            color="secondary"
            size="large"
            keepSizeOnMobile
            className={classes.startingCTA}
            dataTest="about-us-starting-cta"
          >
            {intl.formatMessage({ id: 'ABOUT_US.START_WITH_COMPANY.CTA' })}
          </Button>
        </Link>
      </div>
      <Footer />
    </Layout>
  );
};
export default memo(AboutUs);
