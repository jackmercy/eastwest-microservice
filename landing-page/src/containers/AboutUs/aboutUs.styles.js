import { createUseStyles } from 'react-jss';

import COLORS from '../../constants/COLORS';
import { hexToRgbA, rem } from '../../utils/jss';
import BREAKPOINTS from '../../constants/BREAKPOINTS';

export default createUseStyles({
  primary: { color: COLORS.primary },
  teaser: { textAlign: 'center', paddingTop: rem(32) },
  teaserTitle: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(64),
    lineHeight: rem(88),
  },
  teaserDescription: {
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(40),
    margin: [rem(24), 'auto', 0],
  },
  coverImage: {
    width: '100%',
    height: rem(586),
    marginTop: rem(96),
    backgroundImage: 'url(/static/images/about-us/teaser.png)',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
  },
  menuScrollableLayout: { paddingTop: rem(120), paddingBottom: rem(96) },
  introduceTitle: {
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    cursor: 'pointer',
    '&:nth-of-type(1)': { marginBottom: rem(24), marginTop: 0 },
    '&:nth-of-type(2)': { marginBottom: rem(40), marginTop: rem(160) },
    '&:nth-of-type(3)': { marginBottom: rem(64), marginTop: rem(160) },
  },
  aboutText: { fontSize: rem(24), lineHeight: rem(40) },
  aboutCompany: {
    marginTop: rem(72),
    marginBottom: rem(32),
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    boxSizing: 'border-box',
    backgroundColor: COLORS.bgGray,
    padding: [rem(64), rem(80), rem(109)],
    borderLeft: `1px solid ${COLORS.primary}`,
  },
  aboutCompanySummary: {
    flex: '50%',
    '&:nth-child(3), &:nth-child(4)': { marginTop: rem(85) },
  },
  aboutCompanyValue: {
    fontWeight: 600,
    fontSize: rem(48),
    lineHeight: rem(56),
    marginTop: 0,
    marginBottom: rem(12),
  },
  aboutCompanyLabel: { margin: 0, fontSize: rem(16), lineHeight: rem(24) },
  historyCompany: {},
  historyMark: {
    zIndex: 1,
    display: 'inline-block',
    width: rem(6),
    height: rem(6),
    borderRadius: rem(1),
    verticalAlign: 'middle',
    backgroundColor: COLORS.primary,
    transform: 'rotate(45deg)',
  },
  historySummary: {
    display: 'flex',
    '&:first-child > $historyYear': {
      '&::after': { top: rem(16) },
      '& > $historyMark': {
        width: rem(6),
        height: rem(6),
        right: rem(-4),
        '&::before, &::after': {
          content: '""',
          position: 'absolute',
          top: '50%',
          left: '50%',
          transform: 'translate(-50%, -50%)',
        },
        '&::before': {
          width: rem(16),
          height: rem(16),
          borderRadius: rem(2),
          backgroundColor: hexToRgbA(COLORS.primary, 0.1),
        },
        '&::after': {
          width: rem(24),
          height: rem(24),
          borderRadius: rem(3),
          backgroundColor: hexToRgbA(COLORS.primary, 0.05),
        },
      },
    },
    '&:last-child > $historyDescriptions': { paddingBottom: rem(24) },
  },
  historyYear: {
    paddingRight: rem(68),
    marginRight: rem(113),
    fontWeight: 900,
    fontSize: rem(18),
    lineHeight: rem(32),
    position: 'relative',
    '&::after': {
      content: '""',
      position: 'absolute',
      width: 1,
      height: '100%',
      backgroundColor: COLORS.borderGray,
      right: rem(-2),
      top: 0,
    },
    '& > $historyMark': {
      width: rem(4),
      height: rem(4),
      position: 'absolute',
      top: rem(12),
      right: rem(-3.25),
    },
  },
  historyDescriptions: {
    flex: 1,
    margin: 0,
    listStyleType: 'none',
    padding: [0, 0, rem(76)],
  },
  historyItem: {
    fontSize: rem(18),
    lineHeight: rem(32),
    position: 'relative',
    paddingLeft: rem(20),
    '& + $historyItem': { marginTop: rem(24) },
    '& > $historyMark': {
      position: 'absolute',
      top: rem(12),
      left: 0,
    },
  },
  members: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
  },
  member: {
    width: rem(205),
    textAlign: 'center',
    flex: `${100 / 3}%`,
    marginBottom: rem(64),
  },
  memberAvatar: {
    width: rem(205),
    height: rem(205),
    margin: [0, 'auto'],
    borderRadius: '50%',
    overflow: 'hidden',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
  },
  memberName: {
    marginTop: rem(24),
    marginBottom: 0,
    fontWeight: 900,
    fontSize: rem(24),
    lineHeight: rem(32),
  },
  memberRole: {
    margin: 0,
    fontWeight: 'bold',
    fontSize: rem(16),
    lineHeight: rem(24),
    color: COLORS.primary,
  },
  starting: {
    width: '100%',
    height: rem(360),
    backgroundImage: 'url(/static/images/about-us/get-start-bg.png)',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  startingTitle: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    color: COLORS.textWhite,
  },
  startingCTA: { marginLeft: rem(80) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    teaserTitle: { fontSize: rem(40), lineHeight: rem(48) },
    teaserDescription: {
      fontSize: rem(16),
      lineHeight: rem(24),
      marginTop: rem(16),
    },
    coverImage: {
      marginTop: rem(48),
      backgroundImage: 'url(/static/images/about-us/teaser-mobile.png)',
    },
    menuScrollableLayout: { paddingTop: rem(48) },
    introduceTitle: {
      textAlign: 'center',
      fontSize: rem(40),
      lineHeight: rem(48),
      '&:nth-of-type(1)': { marginBottom: rem(16) },
      '&:nth-of-type(2)': { marginBottom: rem(48), marginTop: rem(64) },
      '&:nth-of-type(3)': { marginBottom: rem(48), marginTop: rem(64) },
    },
    aboutText: { textAlign: 'center', fontSize: rem(16), lineHeight: rem(24) },
    aboutCompany: { marginTop: rem(40), padding: rem(40) },
    aboutCompanySummary: {
      '&:nth-child(3), &:nth-child(4)': { marginTop: rem(64) },
    },
    aboutCompanyValue: {
      fontSize: rem(32),
      lineHeight: rem(40),
      marginBottom: rem(8),
    },
    aboutCompanyLabel: { fontSize: rem(12), lineHeight: rem(16) },
    historyYear: { paddingRight: rem(29), marginRight: rem(29) },
    historyDescriptions: { paddingBottom: rem(34) },
    historyItem: {
      fontSize: rem(16),
      lineHeight: rem(24),
      paddingLeft: 0,
      '& + $historyItem': { marginTop: rem(16) },
      '& > $historyMark': { display: 'none' },
    },
    member: { flex: '100%' },
    starting: {
      flexDirection: 'column',
      height: rem(280),
      backgroundImage: 'url(/static/images/about-us/get-start-bg-mobile.png)',
    },
    startingTitle: { fontSize: rem(32), lineHeight: rem(48) },
    startingCTA: {
      padding: [0, rem(32)],
      marginLeft: 0,
      marginTop: rem(24),
      boxShadow: '2px 4px 12px rgba(0, 0, 0, 0.15)',
    },
  },
});
