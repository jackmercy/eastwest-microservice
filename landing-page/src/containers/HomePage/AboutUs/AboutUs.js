import { memo, useMemo, useState } from 'react';
import { useIntl } from 'react-intl';
import CountUp from 'react-countup';
import VisibilitySensor from 'react-visibility-sensor';

import { Content } from '../../components';

import useStyles from './aboutUs.styles';

const AboutUs = () => {
  const intl = useIntl();
  const classes = useStyles();
  const companyInfo = useMemo(
    () => [
      {
        key: 'founded',
        value: 2018,
        countUpConfigs: { prefix: '', suffix: '' },
        image: '/static/images/homepage/about-us-1.png',
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.FOUNDED',
        }),
        dataTest: 'about-us-info-founded',
      },
      {
        key: 'talents',
        value: 800,
        countUpConfigs: { prefix: '', suffix: '+' },
        image: '/static/images/homepage/about-us-2.png',
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.TALENTS',
        }),
        dataTest: 'about-us-info-talents',
      },
      {
        key: 'projects',
        value: 140,
        countUpConfigs: { prefix: '', suffix: '+' },
        image: '/static/images/homepage/about-us-3.png',
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.PROJECTS',
        }),
        dataTest: 'about-us-info-projects',
      },
      {
        key: 'raised',
        value: 1.5,
        countUpConfigs: {
          prefix: '',
          suffix: 'B+',
          decimals: 1,
          decimal: '.',
        },
        image: '/static/images/homepage/about-us-4.png',
        label: intl.formatMessage({
          id: 'ABOUT_US.DETAILS.ABOUT_COMPANY.RAISED',
        }),
        dataTest: 'about-us-info-raised',
      },
    ],
    [intl],
  );
  const [isVisible, setVisibility] = useState(false);

  const handleVisibilityChange = (status) => {
    if (status) {
      setVisibility(status);
    }
  };

  return (
    <Content className={classes.root}>
      <h2 className={classes.title} data-test="about-us-title">
        {intl.formatMessage({ id: 'HOMEPAGE.ABOUT_US.TITLE' })}
      </h2>
      <p className={classes.description} data-test="about-us-description">
        {intl.formatMessage(
          { id: 'HOMEPAGE.ABOUT_US.DESCRIPTION' },
          { br: <br /> },
        )}
      </p>
      <div className={classes.group}>
        {companyInfo.map((item) => (
          <div
            key={item.key}
            className={classes.groupItem}
            data-test={item.dataTest}
          >
            <img
              alt={item.label}
              src={item.image}
              loading="lazy"
              className={classes.groupItemImage}
            />
            <h2 className={classes.groupItemValue}>
              <CountUp
                end={item.value}
                start={isVisible ? 0 : null}
                {...item.countUpConfigs}
              >
                {({ countUpRef }) => (
                  <VisibilitySensor
                    onChange={handleVisibilityChange}
                    delayedCall
                  >
                    <span ref={countUpRef} />
                  </VisibilitySensor>
                )}
              </CountUp>
            </h2>
            <p className={classes.groupItemLabel}>{item.label}</p>
          </div>
        ))}
      </div>
    </Content>
  );
};

export default memo(AboutUs);
