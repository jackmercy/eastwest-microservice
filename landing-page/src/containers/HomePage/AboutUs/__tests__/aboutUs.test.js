import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AboutUs from '../AboutUs';

describe('Testing HomePage > AboutUs', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<AboutUs />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<AboutUs />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="about-us-title"]')).toHaveLength(1);
  });

  it('should have a description', () => {
    expect(wrapper.find('[data-test="about-us-description"]')).toHaveLength(1);
  });

  it('should show founed information', () => {
    expect(wrapper.find('[data-test="about-us-info-founded"]')).toHaveLength(1);
  });

  it('should show talents information', () => {
    expect(wrapper.find('[data-test="about-us-info-talents"]')).toHaveLength(1);
  });

  it('should show projects information', () => {
    expect(wrapper.find('[data-test="about-us-info-projects"]')).toHaveLength(
      1,
    );
  });

  it('should show raised information', () => {
    expect(wrapper.find('[data-test="about-us-info-raised"]')).toHaveLength(1);
  });
});
