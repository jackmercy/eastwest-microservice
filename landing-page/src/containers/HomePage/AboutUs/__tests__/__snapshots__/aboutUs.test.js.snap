// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage > AboutUs should render match snapshot 1`] = `
<Memo(Content)
  className="root-0-2-1"
>
  <h2
    className="title-0-2-2"
    data-test="about-us-title"
  >
    About us
  </h2>
  <p
    className="description-0-2-3"
    data-test="about-us-description"
  >
    We provide the tools you need to build and scale your business & product.
    <br
      key=".1"
    />
    Get support and guidance to enable Growth and innovation, free from bad practices, risk and technical debt.
  </p>
  <div
    className="group-0-2-4"
  >
    <div
      className="groupItem-0-2-5"
      data-test="about-us-info-founded"
      key="founded"
    >
      <img
        alt="Founded"
        className="groupItemImage-0-2-6"
        loading="lazy"
        src="/static/images/homepage/about-us-1.png"
      />
      <h2
        className="groupItemValue-0-2-7"
      >
        <CountUp
          decimal="."
          decimals={0}
          delay={null}
          duration={null}
          easingFn={null}
          end={2018}
          formattingFn={null}
          onEnd={[Function]}
          onPauseResume={[Function]}
          onReset={[Function]}
          onStart={[Function]}
          onUpdate={[Function]}
          prefix=""
          preserveValue={false}
          redraw={false}
          separator=""
          start={null}
          startOnMount={true}
          suffix=""
          useEasing={true}
        >
          <Component />
        </CountUp>
      </h2>
      <p
        className="groupItemLabel-0-2-8"
      >
        Founded
      </p>
    </div>
    <div
      className="groupItem-0-2-5"
      data-test="about-us-info-talents"
      key="talents"
    >
      <img
        alt="XP Talents"
        className="groupItemImage-0-2-6"
        loading="lazy"
        src="/static/images/homepage/about-us-2.png"
      />
      <h2
        className="groupItemValue-0-2-7"
      >
        <CountUp
          decimal="."
          decimals={0}
          delay={null}
          duration={null}
          easingFn={null}
          end={800}
          formattingFn={null}
          onEnd={[Function]}
          onPauseResume={[Function]}
          onReset={[Function]}
          onStart={[Function]}
          onUpdate={[Function]}
          prefix=""
          preserveValue={false}
          redraw={false}
          separator=""
          start={null}
          startOnMount={true}
          suffix="+"
          useEasing={true}
        >
          <Component />
        </CountUp>
      </h2>
      <p
        className="groupItemLabel-0-2-8"
      >
        XP Talents
      </p>
    </div>
    <div
      className="groupItem-0-2-5"
      data-test="about-us-info-projects"
      key="projects"
    >
      <img
        alt="Amazing Projects"
        className="groupItemImage-0-2-6"
        loading="lazy"
        src="/static/images/homepage/about-us-3.png"
      />
      <h2
        className="groupItemValue-0-2-7"
      >
        <CountUp
          decimal="."
          decimals={0}
          delay={null}
          duration={null}
          easingFn={null}
          end={140}
          formattingFn={null}
          onEnd={[Function]}
          onPauseResume={[Function]}
          onReset={[Function]}
          onStart={[Function]}
          onUpdate={[Function]}
          prefix=""
          preserveValue={false}
          redraw={false}
          separator=""
          start={null}
          startOnMount={true}
          suffix="+"
          useEasing={true}
        >
          <Component />
        </CountUp>
      </h2>
      <p
        className="groupItemLabel-0-2-8"
      >
        Amazing Projects
      </p>
    </div>
    <div
      className="groupItem-0-2-5"
      data-test="about-us-info-raised"
      key="raised"
    >
      <img
        alt="Raised by our clients globally"
        className="groupItemImage-0-2-6"
        loading="lazy"
        src="/static/images/homepage/about-us-4.png"
      />
      <h2
        className="groupItemValue-0-2-7"
      >
        <CountUp
          decimal="."
          decimals={1}
          delay={null}
          duration={null}
          easingFn={null}
          end={1.5}
          formattingFn={null}
          onEnd={[Function]}
          onPauseResume={[Function]}
          onReset={[Function]}
          onStart={[Function]}
          onUpdate={[Function]}
          prefix=""
          preserveValue={false}
          redraw={false}
          separator=""
          start={null}
          startOnMount={true}
          suffix="B+"
          useEasing={true}
        >
          <Component />
        </CountUp>
      </h2>
      <p
        className="groupItemLabel-0-2-8"
      >
        Raised by our clients globally
      </p>
    </div>
  </div>
</Memo(Content)>
`;
